frontend/src/metabase/status/components/FileUploadStatus/FileUploadStatus.tsx

```diff
@@ -49,19 +70,27 @@ const FileUploadStatusContent = ({
   );
   const isVisible = useStatusVisibility(isActive);
 
-  const { data: collection, isLoading } = useCollectionQuery({
-    id: collectionId,
+  const { value: entityInfo, loading } = useAsync(async () => {
+    if (tableId) {
+      const table = await MetabaseApi.table_get({ tableId: tableId });
+      return table;
+    }
+
+    if (collectionId) {
+      const collection = await CollectionsApi.get({ id: collectionId });
```

Comment:
I don't love sidestepping entities here, unless that's something we're doing across the application. Could we use the enabled prop for useCollectionQuery and useTableQuery instead?

Reply:
💡 thanks! I did not know about the `enabled` prop!